crates/cheatcodes/spec/src/vm.rs

```diff
@@ -411,10 +411,25 @@ interface Vm {
 
     /// Revert the state of the EVM to a previous snapshot
     /// Takes the snapshot ID to revert to.
-    /// This deletes the snapshot and all snapshots taken after the given snapshot ID.
+    ///
+    /// **Note:** This does not automatically delete the snapshot. To delete the snapshot use `deleteSnapshot`.
     #[cheatcode(group = Evm, safety = Unsafe)]
     function revertTo(uint256 snapshotId) external returns (bool success);
 
+    /// Revert the state of the EVM to a previous snapshot and automatically deletes the snapshots
+    /// Takes the snapshot ID to revert to.
+    #[cheatcode(group = Evm, safety = Unsafe)]
+    function revertToAndDelete(uint256 snapshotId) external returns (bool success);
+
+    /// Removes the snapshot with the given ID created by `snapshot`.
+    /// Takes the snapshot ID to delete.
```

Comment:
I think it'd be useful to document what happens if the snapshot ID passed in does not exist